#52 예매 조회

0hun opened the pull request:
예매 조회 메소드 추가
예매 조회시 좌석 및 노선 정보 조인하여 조회

0hun committed:
#52 예매 조회

예매 조회 메소드 추가
예매 조회시 좌석 및 노선 정보 조인하여 조회

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/AccountController.java:
```diff
@@ -186,4 +189,31 @@ public void passwordInitBinder(WebDataBinder webDataBinder) {
+     */
+    @LoginCheck(userLevel = UserLevel.ADMIN)
+    @GetMapping("/reservations")
+    public ResponseEntity<List<Reservation>> getReservations(@PathVariable long id) {
```
여기에는 PathVariable이 없어보입니다~

전체 URL로 보자면 `/admin/users/{id}/reservations` 이렇게 하면 특정 회원의 예매 목록이라는게 명확해지지 않을까요?

0hun replied:
회원 자신의 예매목록을 조회하는 것이므로 /accounts/reservations가 맞지 않나요?

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/ReservationController.java:
```diff
@@ -0,0 +1,51 @@
+   * @return ResponseEntity<?>
+   */
+  @LoginCheck(userLevel = UserLevel.ADMIN)
+  @GetMapping("/account/{id}")
```
`account`는 단수입니다. 단수에 단수가 속한다가 성립되지 않을 것 같아요~ 위에서 제안드린 구조로 바꿔보시면 어떨까요?

0hun replied:
이전 결제목록, 마일리지 조회 리뷰에서 관리자가 특정 사용자의 결제 목록 및 마일리지 조회하는 url을 account 단수로 했었습니다. 그래서 이와 같은 구조로 한건데, 변경해야할까요?

f-lab-dev replied:
아 이것은 예매에 있는 특정 사용자의 데이터를 조회하는 것이므로 path variable보다는 query string으로 검색하는게 더 좋을 것 같아요~

f-lab-dev reviewed (commented):
여기서도 트래픽을 많이 받았을 때 잘 처리해줄 수 있는 방법이 있을 것 같은데요.
어떤 방식으로 성능과 가용성을 높힐 수 있을지 한번 설계해보고 저한테 말씀해주시면 피드백드리겠습니다~

0hun committed:
#52 리뷰 반영

회원 자신의 아이디를 @LoginUserId로 받아오도록 수정

0hun committed:
#52 리뷰 반영

예매에 있는 특정 사용자의 예매 정보를 조회하는것 이므로, 단수가 단수에 속한다는 계층적 의미보다는 특정 사용자의 예매 정보를 예매 아이디를 통해 조회하는 의미로
query string으로 수정
